Geometry/HGCalCommonData/src/HGCalCellOffset.cc

```diff
@@ -383,13 +508,314 @@ HGCalCellOffset::HGCalCellOffset(double waferSize,
+                                              xMag}};
+
+        for (int i = 0; i < 6; ++i) {
+          offsetPartialX[k][j - HGCalCell::partiaclWaferCellsOffset][i] = tempOffsetX[i];
+          offsetPartialY[k][j - HGCalCell::partiaclWaferCellsOffset][i] = tempOffsetY[i];
+        }
+      } else if (j == HGCalCell::intTrunCell) {                            // Offset for truncated cells
+        double totalArea = (5.0 * sqrt3_ / 4.0) * std::pow(cellX_[k], 2);  // Area of cell without any dead zone
+        double cutArea =
+            cellX_[k] * sqrt3_ * guardRingOffset_;  // Area of inactive region form guardring and other effects
+        cellAreaPartial[k][j - HGCalCell::partiaclWaferCellsOffset] = totalArea - cutArea;
+        double offMag = (((-2.0 / 15.0) * totalArea * cellX_[k]) - ((cellX_[k] - (0.5 * guardRingOffset_)) * cutArea)) /
+                        (cellArea[k][j]);  // Magnitude of offset
+        // (x, y) coordinates of offset for 6 sides of wafer starting from bottom left edge in clockwise direction
+        // offset_x = -Offset_magnitude * sin(30 + 60*i) i in (0-6)
+        // offset_y = -Offset_magnitude * cos(30 + 60*i) i in (0-6)
+        std::array<double, 6> tempOffsetX = {
+            {-0.5 * offMag, -offMag, -0.5 * offMag, 0.5 * offMag, offMag, 0.5 * offMag}};
+        std::array<double, 6> tempOffsetY = {
+            {-sqrt3By2_ * offMag, 0.0, sqrt3By2_ * offMag, sqrt3By2_ * offMag, 0.0, -sqrt3By2_ * offMag}};
+        for (int i = 0; i < 6; ++i) {
+          offsetX[k][j][i] = tempOffsetX[i];
+          offsetY[k][j][i] = tempOffsetY[i];
+        }
+      } else if (j == HGCalCell::intExtCell) {                             //Offset for extended cells
+        double totalArea = (7.0 * sqrt3_ / 4.0) * std::pow(cellX_[k], 2);  // Area of cell without any dead zone
+        double cutArea =
+            cellX_[k] * sqrt3_ * guardRingOffset_;  // Area of inactive region form guardring and other effects
+        cellAreaPartial[k][j - HGCalCell::partiaclWaferCellsOffset] = totalArea - cutArea;
+        double offMag =  // Magnitude of offset
+            (((5.0 / 42.0) * totalArea * cellX_[k]) - ((cellX_[k] - (0.5 * guardRingOffset_))) * (cutArea)) /
+            (cellArea[k][j]);
+        // (x, y) coordinates of offset for 6 sides of wafer starting from bottom left edge in clockwise direction
+        // offset_x = -Offset_magnitude * sin(30 + 60*i) i in (0-6)
+        // offset_y = -Offset_magnitude * cos(30 + 60*i) i in (0-6)
+        std::array<double, 6> tempOffsetX = {
+            {-0.5 * offMag, -offMag, -0.5 * offMag, 0.5 * offMag, offMag, 0.5 * offMag}};
+        std::array<double, 6> tempOffsetY = {
+            {-sqrt3By2_ * offMag, 0.0, sqrt3By2_ * offMag, sqrt3By2_ * offMag, 0.0, -sqrt3By2_ * offMag}};
+        for (int i = 0; i < 6; ++i) {
+          offsetX[k][j][i] = tempOffsetX[i];
+          offsetY[k][j][i] = tempOffsetY[i];
```

Comment:
@Pruthvi-ch , same here

Reply:
Sorry I have gone through the issue, I will make corrections immediately

Reply:
@Pruthvi-ch , any update on this? This is also causing many runtime errors in UBSAN Ibs

- https://cmssdt.cern.ch/SDT/jenkins-artifacts/ubsan_logs/CMSSW_15_0_X_2025-01-17-2300/logs/a1/a1c295598520c43e1575f98d78b8ed61/log
- https://cmssdt.cern.ch/SDT/jenkins-artifacts/ubsan_logs/CMSSW_15_0_X_2025-01-17-2300/logs/01/017a4c8a5fd844f5a501c51ae21eab85/log

Reply:
I have made PR #47148